portfolio/src/main/java/com/google/sps/data/MergeBooks.java

```diff
@@ -0,0 +1,158 @@
+package com.google.sps.data;
+
+import static com.google.common.base.Preconditions.checkNotNull;
+
+import com.google.common.base.Preconditions;
+import com.google.common.collect.ImmutableList;
+import com.google.sps.data.Book;
+import java.util.ArrayList;
+import java.util.LinkedHashSet;
+import java.util.Set;
+
+public class MergeBooks {
+  /**
+   * This method merges two books that have the same title
+   * and ISBN and returns a new book combined from both of them
+   * @param original: the original book
+   * @param addition: the book with additional info you
+   * want to add
+   * The order of the params does not matter
+   */
+  public static Book merge(Book original, Book addition) {
```

Comment:
When merging books, keep the source (good-reads vs google api) information, and use that for validation.

For example, goodreads data doesn't have ISBN number, but the Google api data does have, so you can check the data coming from different source, and validate if the fields that you expect to be present are actually present in the data source or not. Doing this in generic manner seem to make the code a lot more complicated than it needs to be.

```
public static Book merge(Book googleApiBook, Book goodReadsBook) {

    Precondition.checkArgument(!googleBooksBook.isbn().equals(""), "Expected Google book-api based Book to contain the ISBN-13 populated.");
...
}
```